docs/content/documentation/modules/phase_field/Multi Phase/KKS Derivations.md

```diff
@@ -0,0 +1,362 @@
+\end{eqnarray*}
+$$
+
+Similarly for $c_b$,
+$$
+\begin{eqnarray*}
+J &=& \phi_j \frac{\partial}{\partial c_b} \left( \frac{dh}{d\eta}\frac{dF_a}{dc_a}(c_a-c_b) \right)\\
+&=& -\frac{dh}{d\eta} \phi_j  \frac{d F_a}{d c_a}\\
+\end{eqnarray*}
+$$
+
+For any variable other than $c_a$ or $c_b$, for example temperature $T$:
+
+$$
+\begin{eqnarray*}
+J &=& \phi_j \frac{\partial}{\partial T} \left( \frac{dh}{d\eta}\frac{dF_a}{dc_a}(c_a-c_b) \right)\\
+&=& \frac{dh}{d\eta} \phi_j  \frac{\partial}{\partial T}\left(\frac{d F_a}{d c_a}\right) ( c_a - c_b)\\
+\end{eqnarray*}
+$$
+
+
+The off-diagonal Jacobian contributions are again multiplied by the Allen-Cahn mobility $L$ at each point for consistency with the other terms in the Allen-Cahn equation.
+
+# Constraint Kernels
+
+## ```KKSPhaseChemicalPotential```
+
+This Kernel enforces the point wise equality of the phase chemical potentials
+
+$$
+\frac{dF_a}{dc_a}=\frac{dF_b}{dc_b}.
+$$
+
+The non-linear variable of this Kernel is $c_a$.
+
+### Residual
+
+$$
+R=\frac{dF_a}{dc_a} - \frac{dF_b}{dc_b}
+$$
+
+### Jacobian
+
+For the Jacobian we need to calculate
+
+$$
+J=\frac \partial{\partial u_j}\left( \frac{dF_a}{dc_a} - \frac{dF_b}{dc_b} \right).
+$$
+
+#### On-Diagonal
+
+$$
+J = \phi_j \left( \frac{\partial^2 F_a}{\partial c_a^2} - \frac{\partial^2 F_b}{\partial c_a \partial c_b} \right)
+$$
+
+#### Off-Diagonal
+
+With $q$ the union of the argument vectors of $F_a$ and $F_b$ (represented in the code by ```_coupled_moose_vars[]```) we get
+
+$$
+\sum_i \left( \frac{\partial^2 F_a}{\partial c_a \partial q_i}\frac{\partial q_i}{\partial u_j} - \frac{\partial^2 F_b}{\partial c_b \partial q_i}\frac{\partial q_i}{\partial u_j} \right).
+$$
+
+Again the $\frac{\partial q_i}{\partial u_j}$ is non-zero only if $u\equiv q_i$, which is the case if $q_i$ is the argument selected through ```jvar```.
+
+$$
+J = \frac{\partial^2 F_a}{\partial c_a \partial q_\text{jvar}}\phi_j - \frac{\partial^2 F_b}{\partial c_b \partial q_\text{jvar}}\phi_j.
+$$
+
+Note that in the code ```jvar``` is not an index into ```_coupled_moose_vars[]``` but has to be resolved through the ```_jvar_map```.
+
+## ```KKSPhaseConcentrations```
+
+This Kernel enforces the split of the concentration into the phase concentrations, weighted by the switching function. The non-linear variable of this Kernel is $c_b$.
+
+$$
+c = [1-h(\eta)]c_a+h(\eta)c_b
+$$
+
+### Residual
+
+$$
+R=[1-h(\eta)]c_a + h(\eta)c_b - c
+$$
+
+### Jacobian
+
+#### On-Diagonal
+
+Since the non-linear variable is $c_b$,
+$$
+J= \phi_j \frac{\partial R}{\partial c_b} = \phi_j h(\eta)
+$$
+
+#### Off-Diagonal
+For $c_a$
+$$
+J= \phi_j \frac{\partial R}{\partial c_a} = \phi_j [1-h(\eta)]
+$$
+For $c$
+$$
+J= \phi_j \frac{\partial R}{\partial c} = -\phi_j
+$$
+For $\eta$
+$$
+J= \phi_j \frac{\partial R}{\partial \eta} = \phi_j \frac{dh}{d\eta}(c_b-c_a)
+$$
+
+## Bibliography
+* [KKS] S.G. Kim, W.T. Kim, and T. Suzuki, Phys. Rev. E 60 (1999) 7186
```

Comment:
Utilize the bibtex feature. http://www.mooseframework.org/docs/moose_docs/site/utilities/documentation/moose_flavored_markdown/index.html